### PR TITLE
Add CI to Build and Publish and Update Documentation

### PR DESCRIPTION
**Pull request by mrtz-j**
## Description

This PR makes some additions and updates to the library. It adds GitHub Actions to the build and publish process, updates the README with proper documentation, and adds more examples to the `SampleApp`.

It uses the available NuGet ENV-variables to publish it on NuGet under the name `Giraffe.OpenApi`.

### CODE REVIEW

File: .github/workflows/publish.yml
Comment (64J0) [inline .github/workflows/publish.yml:21]: 👀

File: README.md
Comment (64J0) [inline README.md:5]: Great README!

**Review by 64J0 — APPROVED** [review]
Let's just wait for other people to review the PR too.

File: .github/workflows/publish.yml
Comment (mrtz-j) [reply .github/workflows/publish.yml:21]: I believe it would be beneficial to publish it in the same feed as the other Giraffe packages, but I am open to other suggestions as well.

File: README.md
Comment (dungpa) [inline README.md:32]: The PR looks great.

Is there any chance to merge this and publish a Nuget package version of `Giraffe.OpenApi`?

Comment (64J0) [reply README.md:32]: Sure, I'm just waiting for another maintainer to approve the PR

Comment (dungpa) [reply README.md:32]: @dbrattli @nojaf Can you please help?

**Review by nojaf — APPROVED** [review]
Thank you for this great contribution!

File: README.md
Comment (nojaf) [reply README.md:32]: Hi, sorry I was on holiday. Skimmed over it, looks good from afar.
We can always revisit things later if needed.

**Comment by nojaf** [comment]
Some leftover problem:

image

**Comment by mrtz-j** [comment]
I'll update the restore-targets and try again in another PR :+1:

**Comment by dungpa** [comment]
@mrtz-j @nojaf Is it possible to publish a NuGet package based on this PR?

**Comment by nojaf** [comment]
I believe some yaml tweak is required before publishing will start to work.

**Pull request merged**
